src/msg_recorder/replayer.cc

```diff
@@ -19,7 +19,33 @@ void MessageReplayer::SetSpeedupRate(double rate) {
     speed_up_rate_ = rate;
 }
 
+void MessageReplayer::SetPostStartCallback(std::function<void()>&& post_start) {
+    post_start_ = std::move(post_start);
+}
+
+void MessageReplayer::SetPreFinishCalback(std::function<void()>&& pre_finish) {
+    pre_finish_ = std::move(pre_finish);
+}
+
+void MessageReplayer::SetPostFinishCalback(std::function<void()>&& post_finish) {
+    post_finish_ = std::move(post_finish);
+}
+
 void MessageReplayer::MainLoop() {
+    if (post_start_) {
+        post_start_();
```

Comment:
How about init to an empty func and always call?

Reply:
Does not help, because the caller may pass in an empty function.